spec.html

```diff
@@ -42482,104 +42631,222 @@ <h1>
-          1. Perform LeaveCriticalSection(_WL_) and suspend _W_ for up to _timeout_ milliseconds, performing the combined operation in such a way that a notification that arrives after the critical section is exited but before the suspension takes effect is not lost. _W_ can wake from suspension either because the timeout expired or because it was notified explicitly by another agent calling NotifyWaiter with arguments _WL_ and _W_, and not for any other reasons at all.
+          1. Perform LeaveCriticalSection(_WL_) and suspend the surrounding agent until the time is _waiterRecord_.[[TimeoutTime]], performing the combined operation in such a way that a notification that arrives after the critical section is exited but before the suspension takes effect is not lost. The only way for the surrounding agent to wake from suspension is by another agent calling NotifyWaiter with arguments _WL_ and _thisAgent_. This can happen if the timeout expires or via a call to `Atomics.notify`.
           1. Perform EnterCriticalSection(_WL_).
-          1. If _W_ was notified explicitly by another agent calling NotifyWaiter with arguments _WL_ and _W_, return *true*.
-          1. Return *false*.
+          1. Return ~unused~.
         </emu-alg>
-        <emu-note>
-          <p>_additionalTimeout_ allows implementations to pad timeouts as necessary, such as for reducing power consumption or coarsening timer resolution to mitigate timing attacks. This value may differ from call to call of SuspendAgent.</p>
-        </emu-note>
       </emu-clause>
 
       <emu-clause id="sec-notifywaiter" type="abstract operation">
         <h1>
           NotifyWaiter (
-            _WL_: a WaiterList,
-            _W_: an agent signifier,
+            _WL_: a WaiterList Record,
+            _waiterRecord_: a Waiter Record,
           ): ~unused~
         </h1>
         <dl class="header">
         </dl>
         <emu-alg>
           1. Assert: The surrounding agent is in the critical section for _WL_.
-          1. Notify the agent _W_.
+          1. Assert: _waiterRecord_.[[Result]] is either *"ok"* or *"timed-out"*.
+          1. If _waiterRecord_.[[PromiseCapability]] is ~blocking~, then
+            1. Wake the agent whose signifier is _waiterRecord_.[[AgentSignifier]] from suspension.
+            1. NOTE: This causes the agent to resume execution in SuspendThisAgent.
+          1. Else if AgentSignifier() is _waiterRecord_.[[AgentSignifier]], then
+            1. Let _promiseCapability_ be _waiterRecord_.[[PromiseCapability]].
+            1. Perform ! Call(_promiseCapability_.[[Resolve]], *undefined*, « _waiterRecord_.[[Result]] »).
+          1. Else,
+            1. Perform EnqueueResolveInAgentJob(_waiterRecord_.[[AgentSignifier]], _waiterRecord_.[[PromiseCapability]], _waiterRecord_.[[Result]]).
+            1. NOTE: An agent must not access another agent's promise capability in any capacity beyond passing it to the host.
```

Comment:
I would have probably made this an `emu-note` instead of an inline note, since it applies more generally, at least to these last two branches of the algorithm.

Reply:
Done.